src/app/Main/Body/QueryPage/Results/Charts/BarChartResults.component.tsx

```diff
@@ -0,0 +1,126 @@
+import React from "react";
+import { ApexOptions } from "apexcharts";
+import ReactApexChart from "react-apexcharts";
+import { primaryColor } from "../../../../../Shared/Style/Style-MainTheme.component";
+
+type XYType = { name: string; data: number[] };
+
+function processingJsonApex(
+    JSONTable: Record<string, string>[],
+    xValues: string[]
+): XYType[] {
+    const graphData = [] as XYType[];
+
+    JSONTable.forEach((tableRow) => {
+        const seriesValues: number[] = [];
+        xValues.forEach((element) => {
+            seriesValues.push(Number.parseFloat(tableRow[element]));
+        });
+        const groupData: XYType = { name: tableRow.name, data: seriesValues };
+        graphData.push(groupData);
+    });
+
+    return graphData;
+}
+
+export function BarChartResultsComponent(props: {

```

Comment:
See other comments: This could be `props: { chartData: ChartData}`

Reply:
Done. Commit "Move processing Data to parent & change types"